apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1.IntegrationTests/DatastoreFixture.cs

```diff
@@ -99,5 +128,82 @@ public DatastoreDb CreateDatastoreDb(string namespaceId = null)
             };
             return builder.Build();
         }
+
+        public DatastoreDb CreateDatastoreDbWithDatabase(string databaseId)
+        {
+            var builder = new DatastoreDbBuilder
+            {
+                ProjectId = ProjectId,
+                NamespaceId = NamespaceId,
+                DatabaseId = databaseId ?? "",
+                EmulatorDetection = EmulatorDetection.EmulatorOrProduction
+            };
+            return builder.Build();
+        }
+
+        public async Task RunWithTemporaryDatabaseAsync(Action<string> testFunction)
+        {
+            var databaseId = IdGenerator.FromDateTime(prefix: "test-db-");
+            await CreateDatabaseAsync(databaseId);
+
+            try
+            {
+                testFunction(databaseId);
+            }
+            finally
+            {
+                // Cleanup the test database.
+                try
+                {
+                    await DeleteDatabaseAsync(databaseId);
+                }
+                catch (Exception)
+                {
+                    // Silently ignore errors to prevent tests from failing.
+                }
+            }
+        }
+
+        /// <summary>
+        /// Creates a new Datastore database using Firestore Admin Client.
+        /// </summary>
+        public async Task CreateDatabaseAsync(string databaseId)
+        {
+            var pr = new ProjectName(ProjectId);
+            // Creating a new database is not supported on Datastore emulator.
+            Assert.False(RunningOnEmulator);
+            var client = await GetFirestoreAdminClientAsync();
+            var operation = await client.CreateDatabaseAsync(
+                new ProjectName(ProjectId),
+                new Database { LocationId = "us-east1", Type = DatabaseType.DatastoreMode },
+                databaseId);
+            await operation.PollUntilCompletedAsync(AdminOperationPollSettings);
+            Console.WriteLine($"Success creating database {databaseId}");
+
+        }
+
+        private async Task<FirestoreAdminClient> GetFirestoreAdminClientAsync()
```

Comment:
Is there any reason not to just do this synchronously in the constructor, and expose it as a property?

Reply:
(See https://github.com/googleapis/google-cloud-dotnet/blob/main/apis/Google.Cloud.Firestore/Google.Cloud.Firestore.IntegrationTests/FirestoreFixture.cs for example.)